lib/workers/bin/run_single_worker.rb

```diff
@@ -107,20 +107,14 @@ def all_role_names
   end
 
   update_options = create_options.dup
-  # If a guid is provided, raise if it's not found, update otherwise
-  # Because podified needs to create on the first run_single_worker and update after:
-  #  If system_uid is provided, update if found, create if not found.
-  #  TODO:  This is really inconsistent and confusing.  Why can't GUID follow the same rules?
-  worker = if options[:guid]
-             worker_class.find_by!(:guid => options[:guid]).tap do |wrkr|
-               wrkr.update(update_options)
-             end
-           elsif options[:system_uid] && worker = worker_class.find_by(:system_uid => options[:system_uid])
-             worker.update(update_options)
-             worker
-           else
-             worker_class.create_worker_record(create_options)
-           end
+  find_options   = options.slice(:guid, :system_uid)
```

Comment:
NOTE need to check if either of these ever get passed in as `nil` in which case we would have to drop those

Reply:
Hmm, I'm concerned about this part.  I'd rather get the appliance side to use system_uid and drop guid.  Podified should [always have a system_uid](https://github.com/ManageIQ/manageiq-pods/blob/87d2c0b747a7fcd09961864bb99225baf42a0fe4/images/manageiq-base-worker/container-assets/entrypoint#L16), I think.  I was reluctant to change the appliance side at that time.

Reply:
Yeah that's a great point, we can update to pass in the systemd unit name and drop GUID entirely

Reply:
We are now setting the system_uid for systemd workers as well as of https://github.com/ManageIQ/manageiq/pull/22915